source/common/quic/envoy_quic_client_connection.cc

```diff
@@ -87,6 +87,11 @@ void EnvoyQuicClientConnection::setUpConnectionSocket(Network::ConnectionSocket&
       ENVOY_CONN_LOG(error, "Fail to apply listening options", *this);
       connection_socket.close();
     }
+    const auto maybe_interface_name = connection_socket.connectionInfoProvider().interfaceName();
```

Comment:
This new code is inside the `if (connection_socket.ioHandle().isOpen()) {` guard. Do you understand under what circumstances `setUpConnectionSocket()` would be called but the io handle would not be open? Is this only the failure path?

Reply:
I think the ioctl are failing the testcase, will fix it, thank you for comments.